packages/core/src/scripts/migrate-oz-cli-project.ts

```diff
@@ -0,0 +1,426 @@
+import path from 'path';
+import chalk from 'chalk';
+import { promises as fs } from 'fs';
+import { ManifestData, ImplDeployment } from '../manifest';
+import type { StorageItem, StorageLayout, TypeItem, TypeMembers, StructMember } from '../storage';
+
+const OPEN_ZEPPELIN_FOLDER = '.openzeppelin';
+const EXPORT_FILE = 'openzeppelin-cli-export.json';
+const PROJECT_FILE = path.join(OPEN_ZEPPELIN_FOLDER, 'project.json');
+const SUCESS_CHECK = chalk.keyword('green')('✔') + ' ';
+
+export async function migrateLegacyProject(): Promise<void> {
+  const manifestsMigrationData = await migrateManifestFiles();
+
+  const { compiler } = await getProjectFile();
+  const exportData = {
+    networkFiles: manifestsMigrationData,
+    compiler,
+  };
+
+  await writeJSONFile(EXPORT_FILE, exportData);
+  console.log(SUCESS_CHECK + `Migration data exported to ${EXPORT_FILE}`);
+  await deleteLegacyFiles();
+
+  console.log("\nThese were your project's compiler options:");
+  console.log(compiler);
+}
+
+async function migrateManifestFiles() {
+  const oldManifests = await getManifests();
+  const migratableManifests = oldManifests.filter(manifest => !isDevelopmentNetwork(getNetworkName(manifest)));
+  const { newManifests, manifestsMigrationData } = await migrateManifests(migratableManifests);
+
+  for (const manifestFile in newManifests) {
+    const newManifest = newManifests[manifestFile];
+    const newFileName = getNewManifestLocation(manifestFile);
+    await writeJSONFile(newFileName, newManifest);
+    console.log(SUCESS_CHECK + `Successfully migrated ${newFileName}`);
+  }
+
+  return manifestsMigrationData;
+}
+
+async function deleteLegacyFiles(): Promise<void> {
+  const oldManifests = await getManifests();
+  const developmentManifests = oldManifests.filter(manifestFile => isDevelopmentNetwork(getNetworkName(manifestFile)));
+
+  for (const manifestFile of developmentManifests) {
+    console.log(SUCESS_CHECK + `Deleting unused development manifest ${manifestFile}`);
+    await fs.unlink(manifestFile);
+  }
+
+  console.log(SUCESS_CHECK + `Deleting ${PROJECT_FILE}`);
+  await fs.unlink(PROJECT_FILE);
+}
+
+async function migrateManifests(oldManifests: string[]): Promise<MigrationOutput> {
+  const manifestsMigrationData: Record<string, unknown> = {};
+  const newManifests: Record<string, ManifestData> = {};
+
+  for (const manifestFile of oldManifests) {
+    const oldManifest = JSON.parse(await fs.readFile(manifestFile, 'utf8'));
+
+    if (oldManifest.manifestVersion === undefined) {
+      throw new Error('Migration failed: manifest version too old');
+    }
+
+    const [major, minor] = oldManifest.manifestVersion.split('.');
```

Comment:
I think this should use the same version comparison library we use in `manifest.ts`.